models/operation_manager.go

```diff
@@ -108,55 +98,20 @@ func (o *OperationManager) Start(
 		return err
 	}
 
-	l.Debug("starting check loop on a goroutine")
-	go func() {
-		ticker := time.NewTicker(o.loopTime)
-		defer ticker.Stop()
-
-		for o.continueLoop {
-			select {
-			case <-ticker.C:
-				l.Debug("operationManager loop")
-				wasCanceled, err := o.wasCanceled()
-				if err != nil {
-					l.WithError(err).Warn("error getting operationKey on redis")
-					continue
-				}
-				if wasCanceled {
-					l.Debug("operation was canceled")
-					o.continueLoop = false
-					o.wasCanceledBool = true
-					continue
-				}
-
-				l.Debug("operation was not canceled, continuing...")
-			}
-		}
-	}()
-
 	return nil
 }
 
-// WasCanceled returns a channel that is true when operation was canceled
-func (o *OperationManager) WasCanceled() bool {
+func (o *OperationManager) WasCanceled() (bool, error) {
 	if o == nil {
-		return false
+		return false, nil
 	}
 
-	return o.wasCanceledBool
-}
-
-func (o *OperationManager) wasCanceled() (bool, error) {
 	l := o.logger.WithFields(logrus.Fields{
 		"source":       "OperationManager",
-		"operation":    "wasCanceled",
+		"operation":    "WasCanceled",
 		"operationKey": o.operationKey,
 	})
 
-	if !o.continueLoop {
-		return true, nil
-	}
-
 	l.Debug("checking if operation was canceled")
 	r, err := o.redisClient.HGetAll(o.operationKey).Result()
 	l.Debug("successfully accessed redis")
```

Comment:
This log seems incorrect, maybe we need to log this after checking the `err` variable.

Reply:
Now that you say it does. I didn't change much in this function so I never looked that much into it but will change it.